Sources/HummingbirdCore/Request/RequestBody.swift

```diff
@@ -134,36 +134,84 @@ extension RequestBody {
     /// Delegate for NIOThrowingAsyncSequenceProducer
     @usableFromInline
     final class Delegate: NIOAsyncSequenceProducerDelegate, Sendable {
-        let checkedContinuations: NIOLockedValueBox<Deque<CheckedContinuation<Void, Never>>>
+        enum State {
+            case produceMore
+            case waitingForProduceMore(Deque<CheckedContinuation<Void, Never>>)
+            case terminated
+        }
+        let state: NIOLockedValueBox<State>
 
         @usableFromInline
         init() {
-            self.checkedContinuations = .init([])
+            self.state = .init(.produceMore)
         }
 
         @usableFromInline
         func produceMore() {
-            self.checkedContinuations.withLockedValue {
-                if let cont = $0.popFirst() {
-                    cont.resume()
+            
+            self.state.withLockedValue { state in
+                switch state {
+                case .produceMore:
+                    break
+                case .waitingForProduceMore(var continuations):
+                    if let cont = continuations.popFirst() {
+                        cont.resume()
+                    }
+                    if continuations.count > 0 {
+                        state = .waitingForProduceMore(continuations)
+                    } else {
+                        state = .produceMore
+                    }
```

Comment:
I expect this code is quite performance sensitive, can we restructure to save a copy?

```suggestion
                    switch continuations.count {
                    case 0:
                        assertionFailure("`waitingToProduceMore` state with a continuation count of zero")
                        state = .produceMore
                    case 1:
                        continuations[0].resume()
                        state = .produceMore
                    default:
                        continuations.removeFirst().resume()
                        state = .waitingForProduceMore(continuations)
                    }
```

Reply:
What about having two states, one for a single continuation (The most likely case) and one for multiple continuations.

Reply:
Would mean you wouldn't have the extra allocation for the deque contents

Reply:
That makes sense, my goal was indeed to omit the Deque allocation